cpp/velox/operators/writer/VeloxParquetDatasource.h

```diff
@@ -54,12 +58,14 @@ class VeloxParquetDatasource final : public Datasource {
       const std::string& filePath,
       std::shared_ptr<facebook::velox::memory::MemoryPool> veloxPool,
       std::shared_ptr<facebook::velox::memory::MemoryPool> s3SinkPool,
+      std::shared_ptr<facebook::velox::memory::MemoryPool> gcsSinkPool,
       std::shared_ptr<arrow::Schema> schema)
       : Datasource(filePath, schema),
         filePath_(filePath),
         schema_(schema),
         pool_(std::move(veloxPool)),
-        s3SinkPool_(std::move(s3SinkPool)) {}
+        s3SinkPool_(std::move(s3SinkPool)),
+        gcsSinkPool_(std::move(gcsSinkPool)) {}
```

Comment:
Let's extend the class to VeloxParquetDatsourceS3, VeloxParquetDatsourceGCS and VeloxParquetDatsourceABFS, insted of puting all of them in the same file

Reply:
I agree that the class needs a refactor, but that goes out of the scope of this change.
I propose to make the refactor after this change.

Reply:
I need to onboard `ABFS write` part native support , @FelixYBW / @tigrux , so i make a refactor PR to address this comment https://github.com/apache/incubator-gluten/pull/5486